internal/controller/pipelinerollout/pipelinerollout_controller.go

```diff
@@ -636,40 +629,17 @@ func pipelineObservedGenerationCurrent(generation int64, observedGeneration int6
 func (r *PipelineRolloutReconciler) processPipelineStatus(ctx context.Context, pipelineRollout *apiv1.PipelineRollout, existingPipelineDef *unstructured.Unstructured) error {
 	numaLogger := logger.FromContext(ctx)
 
-	// Only fetch the latest pipeline object while deleting the pipeline object, i.e. when pipelineRollout.DeletionTimestamp.IsZero() is false
-	if existingPipelineDef == nil {
-		// determine name of the promoted Pipeline
-		pipelineName, err := ctlrcommon.GetChildName(ctx, pipelineRollout, r, common.LabelValueUpgradePromoted, nil, r.client, true)
+	if existingPipelineDef != nil {
+		pipelineStatus, err := kubernetes.ParseStatus(existingPipelineDef)
 		if err != nil {
-			return fmt.Errorf("Unable to process pipeline status: err=%s", err)
+			return fmt.Errorf("failed to parse Pipeline Status from pipeline CR: %+v, %v", existingPipelineDef, err)
 		}
-		pipelineDef := &unstructured.Unstructured{}
-		pipelineDef.SetGroupVersionKind(numaflowv1.PipelineGroupVersionKind)
-		pipelineDef.SetNamespace(pipelineRollout.Namespace)
-		pipelineDef.SetName(pipelineName)
 
-		livePipelineDef, err := kubernetes.GetLiveResource(ctx, pipelineDef, "pipelines")
-		if err != nil {
-			if apierrors.IsNotFound(err) {
-				numaLogger.WithValues("pipelineDefinition", *pipelineDef).Warn("Pipeline not found. Unable to process status during this reconciliation.")
-				return nil
-			} else {
-				return fmt.Errorf("error getting Pipeline for status processing: %v", err)
-			}
-		}
-		existingPipelineDef = livePipelineDef
-	}
```

Comment:
This code is not require anymore as child pipeline object is deleted already, so everytime it will fail while getting `livePipelineDef`